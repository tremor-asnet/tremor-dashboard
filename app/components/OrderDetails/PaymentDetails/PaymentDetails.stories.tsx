import type { Meta, StoryObj } from "@storybook/react";

// Components
import PaymentDetails from "./PaymentDetails";

const meta = {
  title: "Components/PaymentDetails",
  component: PaymentDetails,
  tags: ["autodocs"],
} as Meta<typeof PaymentDetails>;

export default meta;

type Story = StoryObj<typeof meta>;

export const PopoverDefault: Story = {
  render: () => <PaymentDetails cardLast4Digit="7852" />,
};
