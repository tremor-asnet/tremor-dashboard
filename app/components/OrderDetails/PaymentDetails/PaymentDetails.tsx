"use client";

import { METADATA } from "@/constants";
// Import libs
import { Flex } from "@tremor/react";
import Image from "next/image";

const PaymentDetails = ({ cardLast4Digit }: { cardLast4Digit: string }) => {
  return (
    <div>
      <h6 className="text-tremor-content-title dark:text-dark-primary font-bold">
        Payment Details
      </h6>
      <Flex
        alignItems="center"
        className="p-6 mt-4 border-[#dee2e6] rounded-lg border">
        <Flex justifyContent="start" alignItems="center">
          <Image
            className="mr-4"
            src={METADATA.MASTERCARD}
            alt="master card icon"
            width="38"
            height="28"
          />
          <h6 className="text-tremor-content-title dark:text-dark-primary font-bold">
            **** **** **** {cardLast4Digit}
          </h6>
        </Flex>
        <div className="group">
          <span className="invisible absolute rounded-md bg-black text-white group-hover:visible p-2 text-xs -translate-x-1/2 translate-y-10 w-42">
            We do not store card details
          </span>
          <button className="w-6 h-6 text-dark-tremor-content-slate bg-transparent border border-dark-tremor-content-slate rounded-full outline outline-2 outline-transparent focus:outline-dark-tremor-content">
            !
          </button>
        </div>
      </Flex>
    </div>
  );
};

export default PaymentDetails;
