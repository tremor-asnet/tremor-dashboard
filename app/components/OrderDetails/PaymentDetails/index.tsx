"use client";

// Components
import { Flex, Bold } from "@tremor/react";
import Image from "next/image";

// Constants
import { METADATA } from "@/constants";

const PaymentDetails = () => {
  return (
    <>
      <Bold className="text-primary font-semibold capitalize dark:text-white tracking-[0.12px]">
        Payment Details
      </Bold>
      <Flex
        alignItems="center"
        className="p-6 mt-4 border-[#dee2e6] rounded-lg border">
        <Flex justifyContent="start" alignItems="center">
          <Image
            className="mr-4"
            src={METADATA.MASTERCARD}
            alt="master card icon"
            width="38"
            height="28"
          />
          <Bold className="text-primary font-semibold capitalize dark:text-white tracking-[0.12px]">
            **** **** **** 7852
          </Bold>
        </Flex>
        <div className="group">
          <span className="invisible absolute rounded-md bg-black text-white group-hover:visible p-2 text-xs -translate-x-1/2 translate-y-10 w-42">
            We do not store card details
          </span>
          <button className="w-6 h-6 text-dark-tremor-content-slate bg-transparent border border-dark-tremor-content-slate rounded-full outline outline-2 outline-transparent focus:outline-dark-tremor-content">
            !
          </button>
        </div>
      </Flex>
    </>
  );
};

export default PaymentDetails;
