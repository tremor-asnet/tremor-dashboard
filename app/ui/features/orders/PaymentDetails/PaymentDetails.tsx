"use client";

import { METADATA } from "@/constants";
// Import libs
import { Flex, Button } from "@tremor/react";
import Image from "next/image";

// Components
import { Popover } from "@/ui/components";

const PaymentDetails = ({ cardLast4Digit }: { cardLast4Digit: string }) => {
  return (
    <div>
      <h6 className="text-tremor-content-title dark:text-dark-primary font-bold">
        Payment Details
      </h6>
      <Flex className="p-6 mt-4 border-[#dee2e6] rounded-lg border">
        <Flex justifyContent="start">
          <Image
            className="mr-4"
            src={METADATA.MASTERCARD}
            alt="master card icon"
            width="38"
            height="28"
          />
          <h6 className="text-tremor-content-title dark:text-dark-primary font-bold">
            **** **** **** {cardLast4Digit}
          </h6>
        </Flex>
        <div className="group relative">
          <Popover
            content="We do not store card details"
            className="text-center !bg-black !bottom-[-50px] min-w-[200px] rounded-md text-white right-[-50px] sm:right-[-60px] lg:right-[-82px] before:content-['â–²'] before:absolute before:top-[-15px] before:right-[54px] sm:before:right-[64px] lg:before:right-[86px] before:text-[black]">
            <Button
              variant="light"
              className="w-6 h-6 text-dark-tremor-content-slate dark:text-dark-tremor-content-slate dark:hover:text-dark-tremor-content-slate bg-transparent border border-dark-tremor-content-slate rounded-full outline outline-2 outline-transparent focus:outline-dark-tremor-content">
              !
            </Button>
          </Popover>
        </div>
      </Flex>
    </div>
  );
};

export default PaymentDetails;
