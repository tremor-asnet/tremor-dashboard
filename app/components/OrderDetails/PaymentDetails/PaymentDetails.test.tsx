import { render } from "@testing-library/react";
import PaymentDetails from ".";

describe("Payment Details Components Testing", () => {
  test("Should render correctly", () => {
    const result = render(<PaymentDetails />);
    expect(result).toBeTruthy();
  });

  test("Should match snapshot", () => {
    const result = render(<PaymentDetails />);
    expect(result).toMatchSnapshot();
  });
});
