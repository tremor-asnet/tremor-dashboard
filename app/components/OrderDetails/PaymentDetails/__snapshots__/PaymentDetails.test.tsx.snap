// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment Details Components Testing Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h6
          class="text-tremor-content-title dark:text-dark-primary font-bold"
        >
          Payment Details
        </h6>
        <div
          class="tremor-Flex-root flex w-full flex-row justify-between items-center p-6 mt-4 border-[#dee2e6] rounded-lg border"
        >
          <div
            class="tremor-Flex-root flex w-full flex-row justify-start items-center"
          >
            <img
              alt="master card icon"
              class="mr-4"
              data-nimg="1"
              decoding="async"
              height="28"
              loading="lazy"
              src="/_next/image?url=%2Fassets%2Fimages%2Fmastercard.webp&w=96&q=75"
              srcset="/_next/image?url=%2Fassets%2Fimages%2Fmastercard.webp&w=48&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fmastercard.webp&w=96&q=75 2x"
              style="color: transparent;"
              width="38"
            />
            <h6
              class="text-tremor-content-title dark:text-dark-primary font-bold"
            >
              **** **** **** 7852
            </h6>
          </div>
          <div
            class="group"
          >
            <span
              class="invisible absolute rounded-md bg-black text-white group-hover:visible p-2 text-xs -translate-x-1/2 translate-y-10 w-42"
            >
              We do not store card details
            </span>
            <button
              class="w-6 h-6 text-dark-tremor-content-slate bg-transparent border border-dark-tremor-content-slate rounded-full outline outline-2 outline-transparent focus:outline-dark-tremor-content"
            >
              !
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h6
        class="text-tremor-content-title dark:text-dark-primary font-bold"
      >
        Payment Details
      </h6>
      <div
        class="tremor-Flex-root flex w-full flex-row justify-between items-center p-6 mt-4 border-[#dee2e6] rounded-lg border"
      >
        <div
          class="tremor-Flex-root flex w-full flex-row justify-start items-center"
        >
          <img
            alt="master card icon"
            class="mr-4"
            data-nimg="1"
            decoding="async"
            height="28"
            loading="lazy"
            src="/_next/image?url=%2Fassets%2Fimages%2Fmastercard.webp&w=96&q=75"
            srcset="/_next/image?url=%2Fassets%2Fimages%2Fmastercard.webp&w=48&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fmastercard.webp&w=96&q=75 2x"
            style="color: transparent;"
            width="38"
          />
          <h6
            class="text-tremor-content-title dark:text-dark-primary font-bold"
          >
            **** **** **** 7852
          </h6>
        </div>
        <div
          class="group"
        >
          <span
            class="invisible absolute rounded-md bg-black text-white group-hover:visible p-2 text-xs -translate-x-1/2 translate-y-10 w-42"
          >
            We do not store card details
          </span>
          <button
            class="w-6 h-6 text-dark-tremor-content-slate bg-transparent border border-dark-tremor-content-slate rounded-full outline outline-2 outline-transparent focus:outline-dark-tremor-content"
          >
            !
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
